src/librustc_resolve/lib.rs

```diff
@@ -2076,7 +2098,7 @@ impl<'a> Resolver<'a> {
                                                        &path,
                                                        trait_ref.path.span,
                                                        trait_ref.path.segments.last().unwrap().span,
-                                                       PathSource::Trait)
+                                                       PathSource::Trait(AliasPossibility::No))
```

Comment:
This function `with_optional_trait_ref` is only invoked when resolving a trait impl or auto trait impl. But there's no documentation so I don't know whether that's intended or not. If not, it's the wrong place to make this distinction.

Reply:
OK, I guess this was tacitly endorsed, so I'll add comments in a future commit.